src/coreclr/tools/Common/Compiler/DisplayNameHelpers.cs

```diff
@@ -36,7 +36,8 @@ public static string GetDisplayName(this MethodDesc method)
 
             if (method.IsConstructor)
             {
-                sb.Append(method.OwningType.GetDisplayNameWithoutNamespace());
+                DefType defType = method.OwningType.GetTypeDefinition() as DefType;
+                sb.Append(defType?.Name ?? method.Name);
```

Comment:
I'll just assume the changes here match what IL Linker is doing. But for the `??` part I want to point out that we would only take that codepath for array constructors (so e.g. `new int[10, 20, 30];`) and whether that part is intentional.

If so, I'd suggest we change the if above to `if (method.IsConstructor && method.OwningType is DefType defType)`. Array constructors will fall down to the last `if` in the cascade.